src/win/util.c

```diff
@@ -1387,3 +1390,146 @@ int uv__getpwuid_r(uv_passwd_t* pwd) {
 int uv_os_get_passwd(uv_passwd_t* pwd) {
   return uv__getpwuid_r(pwd);
 }
+
+
+int uv_getenv(const char* name, char* buffer, size_t* size) {
+  wchar_t var[MAX_ENV_VAR_LENGTH];
+  wchar_t* name_w;
+  DWORD bufsize;
+  size_t len;
+  int r;
+
+  if (name == NULL || buffer == NULL || size == NULL || *size == 0)
+    return UV_EINVAL;
+
+  /* Determine the size of the wide character name */
+  bufsize = MultiByteToWideChar(CP_UTF8, 0, name, -1, NULL, 0);
+
+  if (bufsize == 0)
+    return uv_translate_sys_error(GetLastError());
+
+  /* Convert the environment variable name to a wide character string */
+  name_w = (wchar_t*) uv__malloc(sizeof(wchar_t) * bufsize);
+
+  if (name_w == NULL)
+    return UV_ENOMEM;
+
+  bufsize = MultiByteToWideChar(CP_UTF8, 0, name, -1, name_w, bufsize);
+
+  if (bufsize == 0) {
+    uv__free(name_w);
+    return uv_translate_sys_error(GetLastError());
+  }
+
+  len = GetEnvironmentVariableW(name_w, var, MAX_ENV_VAR_LENGTH);
+  uv__free(name_w);
+
+  if (len == 0) {
+    r = GetLastError();
+
+    if (r == ERROR_ENVVAR_NOT_FOUND)
+      return UV_ENOENT;
+
+    return uv_translate_sys_error(r);
+  } else if (len > MAX_ENV_VAR_LENGTH) {
+    /* This should not be possible */
+    return UV_EIO;
```

Comment:
I'd go for an assert. If we ever see it blow up IRL we can replace it with another error.